R/pillar.R

```diff
@@ -0,0 +1,45 @@
+
+#' @export
+obj_sum.tune_results <- function(x) {
+  null_metrics <- purrr::map_lgl(x$.metrics, is.null)
+  if (!all(null_metrics)) {
+    res <- "tune[+]"
+  } else {
+    res <- "tune[x]"
+  }
+  res
+}
+
+#' Helpers for pillar formatting
+#' @param x an object
+#' @return A character string.
+#' @export
+#' @keywords internal
+#' @rdname pillar-helpers
+size_sum.tune_results <- function(x) {
+  ""
+}
```

Comment:
I am uncomfortable with `@export`ing this

Since you don't do `@importFrom tibble size_sum`, this will export an actual function named `size_sum.tune_results()`. i.e. you will be able to do `tune::size_sum.tune_results()` which we definitely do not want.

I think it you remove the export tag (and all the other documentation) and re-document, then it should still work as expected since we lazy register it in the onLoad instead

To make it clear that this function will be registered in the onLoad, i typically add a comment (not a roxygen comment!) like:

```
# @export - lazy in .onLoad()
```

as done in:
https://github.com/r-lib/clock/blob/56cf09a260e5b9328e18a8fa24f77221191143a8/R/calendar.R#L32

Reply:
It is possible that you will also need to change the name from `size_sum.tune_results` to `size_sum_tune_results`. In rsample, I had to name methods this way when they were being conditionally registered based on a package version https://github.com/tidymodels/rsample/blob/9caeda19ce8fa603d6a42827f8a685b80f573c68/R/compat-dplyr-old.R#L22

It has something to do with the fact that `devtools::load_all()` will always try and register them, since they "look like" S3 methods based on the `.` in their name